src/query.rs

```diff
@@ -762,6 +817,191 @@ macro_rules! tuple_impl {
 //smaller_tuples_too!(tuple_impl, B, A);
 smaller_tuples_too!(tuple_impl, O, N, M, L, K, J, I, H, G, F, E, D, C, B, A);
 
+/// A prepared query can be stored independently of the [`World`] to amortize query set-up costs.
+pub struct PreparedQuery<Q: Query> {
+    memo: (u64, u64),
+    state: Box<[(usize, <Q::Fetch as Fetch<'static>>::State)]>,
+}
+
+impl<Q: Query> Default for PreparedQuery<Q> {
+    fn default() -> Self {
```

Comment:
If this is the normal way for users to construct a `PreparedQuery`, we should also expose a `pub fn new() -> Self` for greater discoverability/concision.